src/domains/sign/api_product/sign/_base_policy.xml

```diff
@@ -1,11 +1,126 @@
 <policies>
     <inbound>
-        <rate-limit-by-key
-            calls="150"
-            renewal-period="5"
-            remaining-calls-header-name="x-rate-limit-remaining" retry-after-header-name="x-rate-limit-retry-after"
-            counter-key="@(context.Subscription.Id)"
-        />
+        <choose>
+            <!-- When called from the outside -->
+            <when condition="@(context.Request.Headers.ContainsKey("X-Forwarded-For"))">
+                <!-- Save the X-Forwarded-For value -->
+                <set-variable
+                    name="originalXForwardedForValue"
+                    value="@(context.Request.Headers.GetValueOrDefault("X-Forwarded-For"))"
+                />
+
+                <!-- Check the client IP value in the X-Forwarded-For header -->
+                <set-header
+                    name="X-Forwarded-For"
+                    exists-action="override"
+                >
+                    <value>@{
+                        int HostToNetworkOrder(int host) {
+                            return (((int)HostToNetworkOrderShort((short)host) & 0xFFFF) << 16)
+                                | ((int)HostToNetworkOrderShort((short)(host >> 16)) & 0xFFFF);
+                        }
+
+                        short HostToNetworkOrderShort(short host) {
+                            return (short)((((int)host & 0xFF) << 8) | (int)((host >> 8) & 0xFF));
+                        }
+
+                        string ipAddress = context.Request.Headers.GetValueOrDefault("x-forwarded-for","");
+                        if (!string.IsNullOrEmpty(ipAddress)) {
+
+                            string[] tokens = ipAddress.Split(':');
+                            if(tokens.Length == 2) {
+                                ipAddress = tokens[0];
+                            }
+
+                            //Place IP Ranges into this list in CIDR notation
+                            // (e.g. "0.0.0.0/0") and separate with commas
+                            List<string> cidrList = new List<string>(){
+                                // Dev Innovation
+                                "51.144.224.93/32",
+                                "51.145.159.13/32",
+                                "52.136.236.15/32",
+                                "52.178.92.96/32",
+                                "52.232.4.132/32",
+                                "52.232.64.246/32",
+                                "52.232.70.142/32",
+
+                                // UniMiB
+                                "51.144.182.201/32",
+                                "149.132.187.0/24",
+
+                                // UniBo
+                                "137.204.0.0/16",
+
+                                // PoliMi
+                                "131.175.148.35/32"
+                            };
```

Comment:
why are they hardcoded?

Reply:
Because the activity of implementing that in cosmos is in progress.
At the moment we can compensate in that way